kernel/src/process_standard.rs

```diff
@@ -1933,6 +1959,10 @@ impl<C: 'static + Chip> ProcessStandard<'_, C> {
     /// this method returns true, the buffer is guaranteed to be readable to the
     /// process.
     fn in_app_flash_memory(&self, buf_start_addr: *const u8, size: usize) -> bool {
+        // TODO: On CHERI platforms, it impossible to form syscalls with pointers
+        // that are not in app memory. However, buf_start_addr is not the right
+        // type to make this function always return true. If CapabilityPtr makes it
+        // slightly further, we can skip this check.
```

Comment:
```suggestion
```

What is CHERI and why is it relevant here?

Reply:
Same as above.